crates/stark-backend/src/air_builders/symbolic/symbolic_expression.rs

```diff
@@ -43,6 +47,36 @@ pub enum SymbolicExpression<F> {
     },
 }
 
+impl<F: Field> Hash for SymbolicExpression<F> {
```

Comment:
Why do we need this? I thought we were just using the raw pointer in the hashmap.

Reply:
This was slightly better because without it, x.clone() * x.clone(), x.clone * x.clone(), record two separate Mul

Reply:
In the unit test, if you do just the pointer hashmap, there are a bunch more nodes